### ISSUE TITLE
Duplicate Transactions (N26 and DKB)

### ISSUE THREAD

**Issue by stockmax** [schurig/ynab-bank-importer#68]
I always get 2 transactions for every transaction. The second one comes one or two days after the first one. This is very confusing as YNAB does not seem to detect the duplicates and I get two transactions and need to delete one of them manually.

Is there a workaorund for the issue? Thank you!
This is such a good project that really helps me with my budget! :)

**Comment by schurig** [comment]
Hi @stockmax, the issue with N26 is knows, however the one with DKB not. Are you sure it also effects DKB transactions?

Unfortunately, there isn't much you can do about it. I introduced a setting some time ago to only import the transaction once it's fully processed. The downside to that setting is that if you enable it, it could take 1-3 days until your transactions show up.

https://github.com/schurig/ynab-bank-importer/wiki/n26#skip_pending_transactions

**Comment by 117agu** [comment]
Hi, just to let you know: I don't get dublicates with DKB. However sometimes my Sparkasse account would show up with transactions "sonstiger Einzug", these are "vorgemerkte Umsätze". Maybe it's something like this for N26? 
I'm just deleting them as they will show up again with the real memo

**Comment by stockmax** [comment]
>Hi @stockmax, the issue with N26 is knows, however the one with DKB not. Are you sure it also effects DKB transactions? ...

@schurig Ah, alright, thanks! I will try the option for N26. 
Regarding DKB: pending transactions are imported and a few days days later the real transactions are imported. YNAB does not recognize the duplicates. Maybe you can introduce an option to ignore pending transactions for DKB (or fints) as well? 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by schurig** [comment]
@stockmax I fear that this would be too specific to a single bank, because there is no DKB-dumper but a generic FinTS dumper that should work for all banks. And I fear if we would match for `sonstiger Einzug` it could skip imports for transactions.. Unless there is some other kind of indication in the memo that is clearer?

**Comment by stockmax** [comment]
@schurig you're right, this should be addressed not only for DKB, but for all banks. I know that with pyhton-fints you can get two xml files (first one with booked transactions, second one with pending transactions). I don't know how this works with ruby_fints though.

**Comment by Technofrikus** [comment]
I have the exact same issue with the DKB. Why cant the solution you found for N26 be applied to the DKB (or other banks)? DKB (or other banks) doesnt provide the necessary information?
